### PR TITLE
Request Authorization, Sanitiation and Validation added

### PR DESCRIPTION
**Pull request by mcuyar**
- Created default Requests and file structure for automating resource authentication, sanitation, and validation
- Created generators and autoloading for resource validators, authorizers, and sanitizers

### CODE REVIEW

File: luminary/Services/Auth/AuthorizesWhenResolvedTrait.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+     * Prepare the instance for authorization.
+     *
+     * @return void
+     */
+    protected function prepareForAuthorization()
+    {
+        // no default action
+    }
+
+
```
Comment (bluewebtech) [inline luminary/Services/Auth/AuthorizesWhenResolvedTrait.php]: Remove unnecessary line break.

File: luminary/Services/Sanitation/Sanitizers/Input.php [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+namespace Luminary\Services\Sanitation\Sanitizers;
+
+class Input
+{
+    /**
+     * Sanitize an input
+     *
+     * @param mixed $input
+     * @param string
```
Comment (bluewebtech) [inline luminary/Services/Sanitation/Sanitizers/Input.php]: Missing param variable.

File: luminary/Services/Sanitation/Sanitizers/Input.php [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+
+        return method_exists($instance, $type) ? $instance->{$type}($input, $arrType) : null;
+    }
+
+     /**
+      * Sanitize an array
+      * $input is array values
+      * $type is type of array values
+      *
+      * @param $input
```
Comment (bluewebtech) [inline luminary/Services/Sanitation/Sanitizers/Input.php]: Missing param type.

File: luminary/Services/Sanitation/Sanitizers/Input.php [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+      * @param $type
```
Comment (bluewebtech) [inline luminary/Services/Sanitation/Sanitizers/Input.php]: Missing param type.

**Comment by bluewebtech** [comment]
@mcuyar There are a couple locations that are missing doc block param information from type to variable. One those have been corrected this can get merged.

**Comment by mcuyar** [comment]
@bluewebtech Changes requested have been pushed.

**Pull request merged**